src/firstboot/firstboot.c

```diff
@@ -866,20 +866,13 @@ static int process_root_args(void) {
                 password = "x";
                 hashed_password = arg_root_password;
         } else if (arg_root_password) {
-                _cleanup_free_ char *salt = NULL;
-                /* hashed_password points inside cd after crypt_r returns so cd has function scope. */
+                r = hash_password(arg_root_password, &_hashed_password);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to encrypt password: %m");
```

Comment:
actually, we really shouldn't call this encryption. It's not encryption. it's hashing. Encryption would mean we could decrypt it, but we cannot, the original version cannot be reconstructed from the hashed password.

(i know this was previously wrong, too, and this doesn't change this, but still the line is modified, hence let's fix "to encryt" → "to hash")